src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3743,12 +3744,42 @@ ImmutableArray<PropertySymbol> addProperties(ImmutableArray<ParameterSymbol> rec
                     if (!memberSignatures.TryGetValue(targetProperty, out var existingMember))
                     {
                         existingMember = OverriddenOrHiddenMembersHelpers.FindFirstHiddenMemberIfAny(targetProperty, memberIsFromSomeCompilation: true);
-                        isInherited = true;
+                        if (existingMember is FieldSymbol)
+                        {
+                            // field from base type should not be considered before we look at fields from current type
+                            existingMember = null;
+                        }
+                        else
+                        {
+                            isInherited = true;
+                        }
+                    }
+
+                    if (existingMember is null)
+                    {
+                        var targetField = new SignatureOnlyFieldSymbol(param.Name,
+                                                                       this,
+                                                                       param.TypeWithAnnotations);
+
+                        if (!memberSignatures.TryGetValue(targetField, out existingMember))
+                        {
+                            existingMember = OverriddenOrHiddenMembersHelpers.FindFirstHiddenMemberIfAny(targetField, memberIsFromSomeCompilation: true);
```

Comment:
Does this method return a different value when called with this targetField than when it is called with the targetProperty above?